src/app/grid/grid-validator-service-cross-cell/grid-validator-service-cross-cell.component.ts

```diff
@@ -0,0 +1,170 @@
+    return (control: AbstractControl): ValidationErrors | null => {
+        let returnObject = {};
+        const createdOnRecord = formGroup.get('created_on');
+        const lastActiveRecord = formGroup.get('last_activity');
+        const winControl = formGroup.get('deals_won');
+        const loseControl = formGroup.get('deals_lost');
+        const actualSalesControl = formGroup.get('actual_sales');
+
+        // Validate dates
+        const curDate = new Date();
+        if (createdOnRecord.value > curDate) {
+            // The created on date shouldn't be greater than current date.
+            returnObject = { ...returnObject, ...{ createdInvalid: { value: createdOnRecord.value } }};
+        }
+        if (lastActiveRecord.value > curDate) {
+            // The last active date shouldn't be greater than current date.
+            returnObject = { ...returnObject, ...{ lastActiveInvalid: { value: lastActiveRecord.value } }};
+        }
+        if (createdOnRecord.value > lastActiveRecord.value) {
+            // The created on date shouldn't be greater than last active date.
+            returnObject = { ...returnObject, ...{ createdLastActiveInvalid: { value: createdOnRecord.value } }};
+        }
+        
+        // Validate deals
+        const dealsRatio = calculateDealsRatio(winControl.value, loseControl.value);
+        if (actualSalesControl.value === 0 && dealsRatio > 0) {
+            returnObject = { ...returnObject, ...{ salesZero: { value: actualSalesControl.value } }};
+        }
+        if (actualSalesControl.value > 0 && dealsRatio == 0 ) {
+            returnObject = { ...returnObject, ...{ salesNotZero: { value: actualSalesControl.value } }};
+        }
+        if (dealsRatio < minDealsRatio) {
+            returnObject = { ...returnObject, ...{ dealsRatio: { value: dealsRatio } }};
+        }
+        
+        return Object.entries(returnObject).length == 0 ? null : returnObject;
+    };
+}
+
+@Directive({
+    selector: '[employeeValid]',
+    providers: [{ provide: NG_VALIDATORS, useExisting: DealsRatioDirective, multi: true }]
+})
+export class DealsRatioDirective extends Validators {
+    @Input('employeeValid')
+    public minDealsRatio = 0;
+
+    public validate(control: AbstractControl): ValidationErrors | null {
+        const formGroup = control.parent;
+        return employeeValidator(this.minDealsRatio, formGroup)(control);
+    }
+}
+
+@Component({
+    selector: 'app-grid-validator-service-cross-cell',
+    styleUrls: ['./grid-validator-service-cross-cell.component.scss'],
+    templateUrl: './grid-validator-service-cross-cell.component.html'
+})
+export class GridValidatorServiceCrossCellComponent {
+
+    @ViewChild('grid1', { read: IgxGridComponent })
+    public grid: IgxGridComponent;
+
+    public minDealsRatio = 0;
+    public transactionData = JSON.parse(JSON.stringify(employeesData));
+
+    constructor() {
+    }
+
+    public formCreateHandler(formGroup: FormGroup) {
```

Comment:
We changed the event arg to IGridFormGroupCreatedEventArgs. This will need update.